modules/standard/FileSystem.chpl

```diff
@@ -1090,6 +1090,76 @@ proc mkdir(name: string, mode: int = 0o777, parents: bool=false) {
 }
 
 pragma "no doc"
+proc moveDir(out error: syserr, src: string, dest: string) {
+  var destExists = exists(error, dest);
+  // Did some error occurred in checking the existance of dest, perhaps a
+  // permissions error?  If so, return.
+  if (error != ENOERR) then return;
+
+  if (destExists) {
+    // dest already existed
+    var aFile = isFile(error, dest);
+    if (error != ENOERR) then return; // Return error messages as encountered
+    var aDir = isDir(error, dest);
+    if (error != ENOERR) then return;
+
+    if (aFile) {
+      // dest is a file, we can't move src within it!
+      error = ENOTDIR;
+      // Note: Python gives EEXISTS in this case, but I think ENOTDIR is
+      // clearer.
+    } else if (aDir) {
+      if (sameFile(src, dest)) {
+        // Python's behavior when calling move over the same directory for
+        // source and destination is to fail with a helpful error message.
+        // Since this error code shouldn't occur otherwise, it signals to
+        // the wrapper function what has happened.
+        error = EEXIST;
+      } else {
+        // dest is a directory, we'll copy src inside it
+        error = EISDIR;
+        // NOT YET SUPPORTED.  Requires basename and joinPath
+      }
+    } else {
+      // What we've been provided is both not a file and not a directory.  Given
+      // the expected behavior of isFile and isDir when it comes to symlinks,
+      // I'm not sure how this case would arise.
+      error = ENOTDIR;
+    }
+  } else {
+    copyTree(error, src, dest, true);
+    if (error != ENOERR) then return; // Error when copying into dest.
+    rmTree(error, src);
```

Comment:
If the copyTree succeeds but the rmTree fails this will make a copy of the directory instead of moving it.  Is that OK?  One way this could happen is if you have read and execute permission on the directory but not write permission.  I experimented with the command line 'mv' utility and it matches this behavior, so I think this is fine.